src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.h

```diff
@@ -0,0 +1,99 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "thread-br-delegate.h"
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/AttributeValueEncoder.h>
+#include <app/CommandHandler.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/ConcreteCommandPath.h>
+#include <inet/UDPEndPoint.h>
+#include <lib/core/Optional.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ThreadBorderRouterManagement {
+
+class ServerInstance : public CommandHandlerInterface, public AttributeAccessInterface, public Delegate::ActivateDatasetCallback
+{
+public:
+    using Status = Protocols::InteractionModel::Status;
+    ServerInstance(EndpointId endpointId, Delegate * delegate) :
+        CommandHandlerInterface(Optional<EndpointId>(endpointId), Id),
+        AttributeAccessInterface(Optional<EndpointId>(endpointId), Id), mDelegate(delegate)
+    {}
+    virtual ~ServerInstance() = default;
+
+    CHIP_ERROR Init();
+
+    // CommandHanlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+
+    // AttributeAccessInterface
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    // ActivateDatasetCallbackInterface
+    void OnActivateDatasetComplete(uint32_t sequenceNum, CHIP_ERROR error) override;
+
+private:
+    friend class TestThreadBorderRouterManagementCluster;
```

Comment:
Friend classes are an anti-pattern for tests. There are ways to split the business logic not to require friends.

Reply:
Add a TODO tag here